proposed/2021/VisualStudioDeepLinkNuGetPackageInstall.md

```diff
@@ -0,0 +1,66 @@
+
+## Solution
+
+For this feature, I will be splitting the work into two main sections, the first being the section that sets up and parses the URI and the second being the logic needed to open the PMUI in Visual Studio.
+
+### URI Work
+
+#### Setting Up URI
+
+Since this feature requires a minimum version of Visual Studio to function, the first step will be to specify to the system what version of Visual Studio to use. I will then register the URI protocol with a .json file to our extension so that Visual Studio may register it. By adding a ProvideAppCommandLineAttribute in the package class, I can then access that URI.
+
+#### Parsing URI
+
+Depending on how the URI link is set up, that link will be split up into its key components that provide me with the information needed to describe a package. Each component will be sub stringed analyzed, and an object will be created with fields associated with said components such as package name and version number. This way, the information in the URI can live somewhere without having to expose the URI in places that it doesn&#39;t need to be.
+
+### Opening the PMUI
+
+#### Opening an Instance Visual Studio When One Isn&#39;t Already Open
+
+(Still looking into ways to implement this) Looking into the code bases to see if there is a programmatic way of externally opening an instance of Visual Studio when the deep link is clicked. Continue to the next section &quot;Opening Solution When an Instance of Visual Studio is Open&quot;.
+
+#### Opening Solution When an Instance of Visual Studio is Open
+
+Instead of the traditional GetToCode window that prompts the user to select a folder or a solution to open, I will investigate offering the user that same GetToCode window experience only this time with only the ability to select solutions as folders cannot access the PMUI. Continue to the next section &quot;Opening Solution Level PMUI From an Open Solution Instance in Visual Studio&quot;.
+
+#### Opening Solution Level PMUI From an Open Solution Instance in Visual Studio
+
+I will invoke opening a ToolWindow that is the same type as the PMUI from an open solution. Continue to the next section &quot;Solution/Project Level PMUI is Open&quot;.
+
+#### Solution/Project Level PMUI is Open
+
+I will first traverse into the &quot;Browse&quot; tab of the PMUI. From there I will programmatically invoke a search for [PackageId: &quot;PackageName&quot;], where ideally one package should show up. Should no package show up, a search for just the package name will be invoked and I will search for an exact name in the results for the target package. Once a package is selected, its package details page will open. From there I&#39;ll invoke setting the package details page to the appropriate version number. From there, it is up to the user to press the install button to finish the process.
+
+## Considerations
```

Comment:
Will you plan to change the package source dropdown? If so, to what? If not, how can you communicate that to users?

Consider that `nuget.org` is probably preferred ~80% of the time, but may not always be enabled/selected in PM UI.

Also consider the ~20% who may want to pass around deep links for packages only on internal feeds.